### PR TITLE
2533 flow precedence with refining

### LINKED ISSUE #2533: Error in flow precedence with refining/extends
OSATE 2.9.0 and 2.7.0

The mode below trigger the following error: 
the source of connection 'C2' does not match the preceding subcomponent or out flow spec feature 'f_pa.f_out'	 on s.impl3
This is rather unexpected, an inspection of the graphical editor seems to confirm the end to end flow is valid.

s.impl2 and s.impl3 are structurally equivalent, the only difference is that one subcomponents of s.impl2 is a refinement of an existing component, the same component in s.impl3 is a rewrite with same interface and flows.

```
package flow_order_test
public

	abstract f_source
		features
			f_so: out feature;
		flows
			flow_so: flow source f_so;
	end f_source;

	abstract f_sink
		features
			f_si: in feature;
		flows
			flow_si: flow sink f_si;
	end f_sink;

	abstract f_path
		features
			f_in: in feature;
			f_out: out feature;
		flows
			flow_pa: flow path f_in -> f_out;

	end f_path;

	system s

	end s;

	system implementation s.impl
		subcomponents
			f_si: abstract f_sink;
			f_so: abstract f_source;
			f_pa: abstract f_path;
		connections
			C1: feature f_so.f_so -> f_pa.f_in;
			C2: feature f_pa.f_out -> f_si.f_si;
		flows

			flow_pa: end to end flow f_so.flow_so -> C1 -> f_pa.flow_pa -> C2 -> f_si.flow_si;
	end s.impl;

	abstract f_path2 extends f_path
		features
			f_in2: in feature;
		flows
			flow_pa2: flow path f_in2 -> f_out;
	end f_path2;

	system implementation s.impl2 extends s.impl
		subcomponents
			f_pa: refined to abstract f_path2;
			f_so2: abstract f_source;
		connections
			C3: feature f_so2.f_so -> f_pa.f_in2;
		flows
			flow_pa2: end to end flow f_so2.flow_so -> C3 -> f_pa.flow_pa2 -> C2 -> f_si.flow_si;
	end s.impl2;

	abstract f_path3 -- f_path3 has exactly the same features and flows as f_path2
		features
			f_in: in feature;
			f_in2: in feature;
			f_out: out feature;
		flows
			flow_pa: flow path f_in -> f_out;
			flow_pa2: flow path f_in2 -> f_out;
	end f_path3;

	system implementation s.impl3 extends s.impl -- s.impl3 should be equivalent to s.impl2
		subcomponents
			f_pa: refined to abstract f_path3;
			f_so2: abstract f_source;
		connections
			C3: feature f_so2.f_so -> f_pa.f_in2;
		flows
			--  error on C2: The source of connection 'C2' does not match the preceding subcomponent or 
			--  out flow spec feature 'f_pa.f_out'	
			flow_pa2: end to end flow f_so2.flow_so -> C3 -> f_pa.flow_pa2 -> C2 -> f_si.flow_si;
	end s.impl3;

end flow_order_test;
```

### PR DESCRIPTION
**Pull request by keh181**
fixes #2533

### CODE REVIEW

File: core/org.osate.core.tests/models/issue2533/flow_order_test.aadl [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	system implementation s.impl3 extends s.impl -- s.impl3 should be equivalent to s.impl2
+		subcomponents
+			f_pa: refined to abstract f_path3;
+			f_so2: abstract f_source;
+		connections
+			C3: feature f_so2.f_so -> f_pa.f_in2;
+		flows
+			--  error on C2: The source of connection 'C2' does not match the preceding subcomponent or 
+			--  out flow spec feature 'f_pa.f_out'	
+			flow_pa2: end to end flow f_so2.flow_so -> C3 -> f_pa.flow_pa2 -> C2 -> f_si.flow_si;
```
Comment (jjhugues) [inline core/org.osate.core.tests/models/issue2533/flow_order_test.aadl:78]: Please update the example. I produced it in the context of the bug report

suggestion: change line 76 to
issue#253 produce the following wrong error message: "..."

Hence, future review of the test will know what the core issue was.

At first reading, one would expect 78 to produce this error message

File: core/org.osate.core.tests/src/org/osate/core/tests/issues/Issue2533Test.xtend [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+//	FluentIssueCollection issues
+	
+	@Test
+	def void test(){
+		val testFileResult = testHelper.testFile(FILE_LOCATION + FILE_NAME)
+		val issueCollection = new FluentIssueCollection(testFileResult.resource, newArrayList, newArrayList)
+		
+		issueCollection.sizeIs(0)
+	}
+}
```
Comment (jjhugues) [inline core/org.osate.core.tests/src/org/osate/core/tests/issues/Issue2533Test.xtend:55]: Should we add a new line here?

**Review by jjhugues — CHANGES_REQUESTED** [review]

File: core/org.osate.core.tests/src/org/osate/core/tests/issues/Issue2533Test.xtend (hunk 2 above)
Comment (keh181) [reply core/org.osate.core.tests/src/org/osate/core/tests/issues/Issue2533Test.xtend:55]: Do you mean that we should remove line 52 or add another assertion for the test?

Comment (jjhugues) [reply core/org.osate.core.tests/src/org/osate/core/tests/issues/Issue2533Test.xtend:55]: No, I meant there should be an empty line 56, as it is a formatting rule in some languages

**Review by jjhugues — APPROVED** [review]

**Pull request merged**